vkrunner/vr-script.c

```diff
@@ -1862,6 +1862,41 @@ read_line(FILE *f,
         return buffer->length > 0;
 }
 
+static const char *
+read_line_from_string(const char *string,
+                      struct vr_buffer *buffer)
```

Comment:
I think this function could be quite a bit cleaner if it scanned for the end of the line once and then added the whole line to the buffer instead of trying to add one character at a time while looking for the end of the line. It could be something like this:

```
        buffer->length = 0;

        const char *end = strchr(string, '\n');

        if (end == NULL) {
                vr_buffer_append_string(buffer, string);
        } else {
                vr_buffer_append(buffer, string, end - string + 1);
                vr_buffer_append_c(buffer, '\0');
                buffer->length--;
        }

        return string + buffer->length;
```

Reply:
Thank you for letting me know it.